k8s/k8s.go

```diff
@@ -35,12 +36,12 @@ func (s *k8sSecrets) GetSecret(
 
 	secret, err := core.Instance().GetSecret(secretName, namespace)
 	if err != nil {
+		if kerrors.IsNotFound(err) {
```

Comment:
Doesn't `"github.com/portworx/sched-ops/k8s/core"` provide the `IsNotFound()` function? It will be good to avoid a new dependency.

Reply:
Where?

The secrets update logic uses `strings.Contains(err.Error(), "not found"` to detect this condition, see [here](https://github.com/portworx/sched-ops/blob/1f213fafc48e8649820ebf683c4d23ce0fc28d99/k8s/core/secrets.go#L75)

Reply:
Ok so they don't have it. I'm sure how the maintainers feel about adding another dependency. At the same time the initial version with `strings.Contains(err.Error(), "not found"` is probably ok and not worth the additional dep.

Reply:
Sounds good, pushing the string detection version. This is the minimal change to fix the logic.

Reply:
It should be ok to add a new dependency. Better to have a strict error check instead of string comparison.

Reply:
Not sure if the new dep is worth it for such a small check though...